### PR TITLE
Create a pro engine connector hook in Constant_propagation

### PR DESCRIPTION
**Pull request by ihji**
test plan:
See semgrep/semgrep-proprietary#1102

### CODE REVIEW

File: src/analyzing/Constant_propagation.ml [hunk 1]
```diff
@@ -623,6 +623,20 @@ let add_special_constants env lang prog =
 (* Entry point *)
```
Comment (IagoAbal) [inline src/analyzing/Constant_propagation.ml:623]: We need a changelog entry to announce that Pro will recognize constants defined via macros.

Comment (IagoAbal) [reply src/analyzing/Constant_propagation.ml:623]: (You should always follow the PR checklist.)

File: src/analyzing/Constant_propagation.mli [hunk 2]
```diff
@@ -6,6 +6,20 @@ val constant_propagation_and_evaluate_literal :
   ?lang:Lang.t -> AST_generic.expr -> AST_generic.svalue option
 (** Partially evaluate a Generic expression. *)
 
+type env
+
+type propagate_basic_visitor_funcs = {
```
Comment (IagoAbal) [inline src/analyzing/Constant_propagation.mli:11]: Maybe add some comments to clarify what we are exporting to be used by Pro.

File: src/analyzing/Constant_propagation.ml [hunk 3]
```diff
@@ -623,6 +623,20 @@ let add_special_constants env lang prog =
...
 (*****************************************************************************)
 
+type propagate_basic_visitor_funcs = {
+  visit_definition :
+    env * Iter_with_context.context -> AST_generic.definition -> unit;
+  visit_expr :
```
Comment (IagoAbal) [inline src/analyzing/Constant_propagation.ml]: This one you're not using it yet it seems, maybe it's better to drop it until we need it?

**Review by IagoAbal — APPROVED** [review]

**Comment by IagoAbal** [comment]
We should always follow the checklist, it is there so that we do not forget about things like adding a test plan.

File: src/analyzing/Constant_propagation.ml (hunk 1 above)
Comment (ihji) [reply src/analyzing/Constant_propagation.ml:623]: Done.

File: src/analyzing/Constant_propagation.mli (hunk 2 above)
Comment (ihji) [reply src/analyzing/Constant_propagation.mli:11]: Done.

File: src/analyzing/Constant_propagation.ml (hunk 3 above)
Comment (ihji) [reply src/analyzing/Constant_propagation.ml]: Done.

**Pull request merged**